identity-server/src/IdentityServer/Validation/Default/PrivateKeyJwtSecretValidator.cs

```diff
@@ -119,9 +136,13 @@ public async Task<SecretValidationResult> ValidateAsync(IEnumerable<Secret> secr
                 return audValue is string audString &&
                        AudiencesMatch(audString, issuer);
             };
+
+            // Strict audience validation requires that the token type be "JWT"
+            tokenValidationParameters.ValidTypes = ["JWT"];
```

Comment:
This is the old type value. With strict validation, we want the new value here: "client-authentication+jwt".